seal.go

```diff
@@ -272,32 +256,25 @@ func SealKeyToTPM(tpm *TPMConnection, key []byte, keyPath, policyUpdatePath stri
 		pcrPolicyCounterPub: pcrPolicyCounterPub,
 		lockIndexName:       lockIndexName})
 	if err != nil {
-		return xerrors.Errorf("cannot compute static authorization policy: %w", err)
+		return nil, xerrors.Errorf("cannot compute static authorization policy: %w", err)
 	}
 
 	// Define the template for the sealed key object, using the computed policy digest
 	template.AuthPolicy = authPolicy
-	sensitive := tpm2.SensitiveCreate{Data: key}
 
-	// Have the digest of the private data recorded in the creation data for the sealed data object.
-	authKeyBytes, err := x509.MarshalECPrivateKey(authKey)
+	// Create the sensitive data
+	sealedData, err := tpm2.MarshalToBytes(sealedData{Key: key, AuthPrivateKey: authKey.D.Bytes()})
 	if err != nil {
-		return xerrors.Errorf("cannot marshal key for signing authorization policy updates: %w", err)
+		panic(fmt.Sprintf("cannot marshal sensitive data: %v", err))
 	}
-
-	h := crypto.SHA256.New()
-	if _, err := tpm2.MarshalToWriter(h, authKeyBytes); err != nil {
-		panic(fmt.Sprintf("cannot marshal dynamic authorization policy update data: %v", err))
-	}
-	creationInfo := h.Sum(nil)
+	sensitive := tpm2.SensitiveCreate{Data: sealedData}
 
 	// Now create the sealed key object. The command is integrity protected so if the object at the handle we expect the SRK to reside
 	// at has a different name (ie, if we're connected via a resource manager and somebody swapped the object with another one), this
 	// command will fail. We take advantage of parameter encryption here too.
-	priv, pub, creationData, _, creationTicket, err :=
-		tpm.Create(srk, &sensitive, template, creationInfo, nil, session.IncludeAttrs(tpm2.AttrCommandEncrypt))
+	priv, pub, _, _, _, err := tpm.Create(srk, &sensitive, template, nil, nil, session.IncludeAttrs(tpm2.AttrCommandEncrypt))
```

Comment:
can you clarify why we can ignore creationTicket and creationData now?

Reply:
We were only using the creation ticket and creation data to cryptographically bind the file containing the authorization key to the sealed key object for the purposes of verifying later on that they were consistent in keyData.validate, but as that has been removed then it's ok to ignore these now.